src/core/Akka/Util/FastLazy.cs

```diff
@@ -81,5 +81,80 @@ public T Value
             }
         }
     }
-}
 
+
+    /// <summary>
+    /// A fast, atomic lazy that only allows a single publish operation to happen,
+    /// but allows executions to occur concurrently.
+    /// 
+    /// Does not cache exceptions. Designed for use with <typeparamref name="T"/> types that are <see cref="IDisposable"/>
+    /// or are otherwise considered to be expensive to allocate. 
+    /// 
+    /// Read the full explanation here: https://github.com/Aaronontheweb/FastAtomicLazy#rationale
+    /// </summary>
+    /// <typeparam name="S">State type</typeparam>
+    /// <typeparam name="T">Value type</typeparam>
+    public sealed class FastLazy<S, T>
```

Comment:
See https://github.com/Rattenkrieg/akka.net/commit/05bf8698fc7468da455b769c147cd1ae10b20ce0#diff-74f21fe653ff44fb831459d0f3120fc0R181
A caching optimization for delegates. But now I need to somehow pass parameters to cached func. Decided to create separate class to keep original `FastLazy` _struct_ slim in terms of memory (additional space for state) and code hassle (overloaded ctor, etc).